src/fsharp/NameResolution.fs

```diff
@@ -1538,14 +1538,18 @@ type TcResultsSinkImpl(g, ?source: string) =
     let formatStringCheckContext =
         lazy
             source |> Option.map (fun source ->
-                let source = source.Replace("\r\n", "\n").Replace("\r", "\n")
                 let positions =
-                    source.Split('\n')
-                    |> Seq.map (fun s -> String.length s + 1)
-                    |> Seq.scan (+) 0
-                    |> Seq.toArray
-                { NormalizedSource = source 
-                  LineEndPositions = positions })
+                    [|
+                        yield 0
```

Comment:
Possible bug: So suppose the file content is just the 1 byte character `\n`. This will give for `positions`:
`[| 0; 1; 1 |]` Should not we want just `[| 0; 1 |]` instead? CC: @forki  @AviAvni